src/modules/core/components/ActionsList/ActionsListItem.tsx

```diff
@@ -75,8 +85,16 @@ const ActionsListItem = ({
    * Address checksum / normalization needs to be fixed on the
    * subgraph side
    */
-  const initiatorUserProfile = useUser(createAddress(initiator));
-  const recipientAddress = createAddress(recipient);
+  const initiatorUserProfile = useUser(
+    initiator ? createAddress(initiator) : '',
+  );
+  const agentUserProfile = useUser(agent ? createAddress(agent) : '');
+  const eventRecipient = values?.recipient
+    ? createAddress(values.recipient)
+    : null;
+  const recipientAddress = recipient
+    ? createAddress(recipient)
+    : null || eventRecipient || colonyAddress;
```

Comment:
what is happening here? if eventRecipient and colonyAddress will be undefined we will get here undefined either way.

So why not just 
: eventRecipient || colonyAddress

Reply:
Huh?

It recipient _or_ eventRecipient _or_ colonyaddress.

Just that the first part needs to check that the recipient actually exists, otherwise it can't call createAddress since it will error out: 
```
recipient
    ? createAddress(recipient)
    : null
```
So if that is null, than go to the the `eventRecipient || colonyAddress` check

Yeah I know it's kinda cumbersome, and this is because we're trying to make the ActionsList component handle events as well. 

I think a better idea would be for me to just create a separate one just for events. That way there would be no need for such "awkward" logic checks everywhere.